File: benchmarks/language_compare/function_calls.py
# BENCH_ITERATIONS=1000000 BENCH_EXPECTED=7270106
def gcd(a, b):
    return a if b == 0 else gcd(b, a % b)
state = 1
checksum = 0
for _ in range(1_000_000):
    state = (state * 48_271) % 2_147_483_647
    checksum = (checksum + gcd(state, state % 10_000 + 12_345)) % 1_000_000_007
print(checksum)

File: benchmarks/language_compare/function_calls.go
// BENCH_ITERATIONS=1000000 BENCH_EXPECTED=7270106
package main
import "fmt"
func gcd(a, b int64) int64 { if b == 0 { return a }; return gcd(b, a%b) }
func main() {
	var state int64 = 1
	var checksum int64
	for i := 0; i < 1000000; i++ {
		state = (state * 48271) % 2147483647
		checksum = (checksum + gcd(state, state%10000+12345)) % 1000000007
	}
	fmt.Println(checksum)
}
